RemoteInput/Plugin/Python/PythonEIOS.cxx

```diff
@@ -54,150 +55,181 @@ PyObject* PyEIOS_Str(PyObject* object)
+    {(char*)"send_string", reinterpret_cast<PyCFunction>(Python_SendString), METH_VARARGS, (char*)""},
+    {(char*)"hold_key", reinterpret_cast<PyCFunction>(Python_HoldKey), METH_VARARGS, (char*)""},
+    {(char*)"release_key", reinterpret_cast<PyCFunction>(Python_ReleaseKey), METH_VARARGS, (char*)""},
+    {(char*)"is_key_held", reinterpret_cast<PyCFunction>(Python_IsKeyHeld), METH_VARARGS, (char*)""},
+    {(char*)"get_keyboard_speed", reinterpret_cast<PyCFunction>(Python_GetKeyboardSpeed), METH_VARARGS, (char*)""},
+    {(char*)"set_keyboard_speed", reinterpret_cast<PyCFunction>(Python_SetKeyboardSpeed), METH_VARARGS, (char*)""},
+    {(char*)"get_keyboard_repeat_delay", reinterpret_cast<PyCFunction>(Python_GetKeyboardRepeatDelay), METH_VARARGS, (char*)""},
+    {(char*)"set_keyboard_repeat_delay", reinterpret_cast<PyCFunction>(Python_SetKeyboardRepeatDelay), METH_VARARGS, (char*)""},
+    {(char*)"get_graphics_scaling", reinterpret_cast<PyCFunction>(Python_GetGraphicsScaling), METH_VARARGS, (char*)""},
+    {(char*)"set_graphics_scaling", reinterpret_cast<PyCFunction>(Python_SetGraphicsScaling), METH_VARARGS, (char*)""},
+    {(char*)"kill_client", reinterpret_cast<PyCFunction>(Python_KillClient), METH_VARARGS, (char*)""},
+    
+    {nullptr}  /* Sentinel */
 };
 #endif
 
 PyType_Slot PyEIOS_Slots[] = {
-        {Py_tp_init, nullptr},
-        {Py_tp_members, PyEIOS_Members},
-        {Py_tp_methods, PyEIOS_Methods},
-        {Py_tp_getset, PyEIOS_PropertyMembers},
-        {Py_tp_dealloc, reinterpret_cast<void*>(&PyEIOS_Dealloc)},
-        {Py_tp_clear, reinterpret_cast<void*>(&PyEIOS_Clear)},
-        {Py_tp_str, reinterpret_cast<void*>(&PyEIOS_Str)},
-        {Py_tp_doc, static_cast<void*>(const_cast<char*>(PyDoc_STR("EIOS Structure")))},
-        {0, NULL},
+    {Py_tp_init, nullptr},
+    {Py_tp_members, PyEIOS_Members},
+    {Py_tp_methods, PyEIOS_Methods},
+    {Py_tp_getset, PyEIOS_PropertyMembers},
+    {Py_tp_dealloc, reinterpret_cast<void*>(&PyEIOS_Dealloc)},
+    {Py_tp_clear, reinterpret_cast<void*>(&PyEIOS_Clear)},
+    {Py_tp_str, reinterpret_cast<void*>(&PyEIOS_Str)},
+    {Py_tp_doc, static_cast<void*>(const_cast<char*>(PyDoc_STR("EIOS Structure")))},
+    {0, NULL},
 };
 
 PyType_Spec PyEIOS_Spec {
-        "remote_input.EIOS",
-        sizeof(PyEIOS),
-        0,
-        #if HAS_PYTHON_VERSION(0x030A0000)
-        Py_TPFLAGS_DEFAULT | Py_TPFLAGS_BASETYPE | Py_TPFLAGS_IMMUTABLETYPE | Py_TPFLAGS_DISALLOW_INSTANTIATION,
-        #else
-        Py_TPFLAGS_DEFAULT | Py_TPFLAGS_BASETYPE,
-        #endif
-        PyEIOS_Slots
+    "remote_input.EIOS",
+    sizeof(PyEIOS),
+    0,
+    #if HAS_PYTHON_VERSION(0x030A0000)
+    Py_TPFLAGS_DEFAULT | Py_TPFLAGS_BASETYPE | Py_TPFLAGS_IMMUTABLETYPE | Py_TPFLAGS_DISALLOW_INSTANTIATION,
+    #else
+    Py_TPFLAGS_DEFAULT | Py_TPFLAGS_BASETYPE,
+    #endif
+    PyEIOS_Slots
 };
 
-// MARK: EXPORTS
+PyTypeObject* PyEIOS_Type;
```

Comment:
Why is this a global variable? If we want to do this, it should be near the top of the file then.

Reply:
In order for Py_IsInstance to work, the type created as part of the module has to be global.

Reply:
I was colocating it with the other PyEIOS defined things.